pkg/xds/generator/admin_proxy_generator.go

```diff
@@ -41,13 +63,39 @@ func (g AdminProxyGenerator) Generate(ctx xds_context.Context, proxy *core_xds.P
 
 	resources := core_xds.NewResourceSet()
 
+	for _, se := range staticEnpointPaths {
+		se.ClusterName = envoyAdminClusterName
+	}
+
+	for _, se := range staticTlsEnpointPaths {
+		se.ClusterName = envoyAdminClusterName
+
+		token, err := ctx.EnvoyAdminClient.GenerateAPIToken(proxy.Dataplane)
+		if err != nil {
+			return nil, err
+		}
+		se.Header = "Authorization"
+		se.HeaderExactMatch = fmt.Sprintf("Bearer %s", token)
```

Comment:
Is it okay that the token is seen in the `config_dump`?

Reply:
I think yes since the token is specific to the particular instance of the DP, so no explicit access to other pods.

Reply:
We have to be careful when we will have functionality to see a config dump in GUI or something.